main.tf

```diff
@@ -85,7 +86,7 @@ resource "google_cloudfunctions_function" "main" {
   labels                = var.labels
   runtime               = var.runtime
   environment_variables = var.environment_variables
-  source_archive_bucket = google_storage_bucket.main.name
+  source_archive_bucket = var.create_bucket ? google_storage_bucket.main[0].name : var.bucket_name
```

Comment:
Do we want to coalesce(var.bucket_name, var.name) here also?

Reply:
I think it makes sense to keep it as is.